transport/http/configure_http2_go116.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2021 Google LLC.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build go1.16
+
+package http
+
+import (
+	"net/http"
+	"time"
+
+	"golang.org/x/net/http2"
+)
+
+// configureHTTP2 configures the ReadIdleTimeout HTTP/2 option for the
+// transport. This allows broken idle connections to be pruned more quickly,
+// preventing the client from attempting to re-use connections that will no
+// longer work.
+func configureHTTP2(trans *http.Transport) {
+	http2Trans, err := http2.ConfigureTransports(trans)
```

Comment:
I might be missing something, but should we be using `ConfigureTransport`? Or is `http2Trans` magically connecting some pointers under the hood. I guess I don't understand how the setting on line 23 will do anything to `trans`, am I missing something?

Reply:
I asked Damien the same question and he confirmed that this is how it's supposed to be used. @neild can you comment on how this works under the hood?

Reply:
`http2.ConfigureTransport` just calls `http2.ConfigureTransports` and discards the `*Transport`.

`ConfigureTransports` is a newer function added to allow modifying the `http2.Transport` configuration.

Reply:
@neild Right, I do understand this provides some toggles to configure settings. What I was trying to ask is should this method,  configureHTTP2, should be returning http2Trans as a RoundTripper? But it look like this transport is embedded into `trans` via [RegisterProtocol](https://pkg.go.dev/net/http#Transport.RegisterProtocol) now that I look a little closer. Would there be any benefit returning  http2Trans though?

Reply:
I verified via local testing with `http2debug=1` that a storage client created using this does, in fact, send a PING every 15s.